### ISSUE TITLE
EKS testing collab with Infra

### ISSUE THREAD

**Issue by laineymajor** [department-of-veterans-affairs/va.gov-team#71586]
## Status
_Update each sprint until completed_
| Date | Status | Launch Date | Notes |
| ----- | ------ | ------------------------- | ------ |
| 4/24/25 | Blocked | TBD | Sprint 26: No new updates, the SRE Team is in a holding pattern pending guidance from the Infrastructure Team detailing what backend work is needed. |
| 4/3/25 | Blocked | TBD | Sprint 24: No new updates, please see below for status. |
| 3/27/25 | Blocked | TBD | Sprint 24: This epic is stale. On 3/17 the SRE Team PM requested that the Infrastructure Team PM set up a meeting with the key developers to discuss what cross-team dependencies and what backend work is needed - given the recent EKS upgrade work tracked in the Automate [EKS Cluster Deployment and Tear Down epic](https://github.com/department-of-veterans-affairs/va.gov-team/issues/50671). This discussion will prompt tickets being handed off by the Infrastructure Team, or tickets being written up by the SRE Team. |
| 3/13/25 | Blocked | TBD | Sprint 23: This epic is stale. It was written over 1 year ago, and the problem statement references an EKS upgrade that was planned for January 2024. This initiative was discussed during the 3/17 engineering workstream meeting, and the SRE Team PM requested that the Infrastructure Team PM set up a meeting with the key developers to discuss what cross-team dependencies and what backend work is needed - given the recent EKS upgrade work tracked in the Automate [EKS Cluster Deployment and Tear Down epic](https://github.com/department-of-veterans-affairs/va.gov-team/issues/50671). This discussion will prompt tickets being handed off by the Infrastructure Team, or tickets being written up by the SRE Team. |
| 3/13/25 | Blocked | TBD | Sprint 23: Given the EKS upgrade work tracked in the Automate [EKS Cluster Deployment and Tear Down epic](https://github.com/department-of-veterans-affairs/va.gov-team/issues/50671), the remaining 5 tickets in this epic may be obsolete and/or different tickets might need to be written up for the backend work that is needed. The SRE Team PM is awaiting instruction regarding the backend responsibility, or new tickets to be handed off by the Infrastructure Team. |
| 2/28/25 | In-progress | End of Q1 2025 | Sprint 22: No major updates this week. Work is still paused in favor of focusing resources on the top priority initiatives the EVSS migration & PCIU migration. |
| 2/21/25 | In-progress | End of Q1 2025 | Sprint 21: No major updates this week. Work is still paused in favor of focusing resources on the top priority initiatives the EVSS migration & PCIU migration. |
| 2/7/25 | In-progress | End of Q1 2025 | Sprint 20: No major updates from last week. The PM's discussed the dependency with the DevOps work. |
| 1/31/25 | In-progress | End of Q1 2025 | Sprint 20: Logs and monitoring testing were completed during Sprint 19, and load testing continues into Sprint 20. No additional work is planned for the current sprint, as the team remains focused on the EVSS migration discovery and supporting the developer working on the PCIU mobile migration (so the remaining Platform work can be unblocked). One challenge was identified this week with the next component that is to be tested - scaling. The team is blocked because there is no available endpoint (there is no ingress to the cluster, so there is no URL to send the request), this is a dependency with the Infrastructure Team because it requires DevOps. I'll bring this topic for discussion during the Engineering Workstream Sync on Monday. |
| 1/24/25 | In-progress | End of Q1 2025 | Sprint 19: After discussion with Platform Leadership, the scope of the remaining testing will be narrowed to include only the manual testing and to remove the portion of the tickets that builds automation. This initiative is not a blocker to the VAEC initiative, but it is a nice-to-have because it supports an EKS upgrade (a separate initiative). By upgrading before the forced upgrade in May 2025, we ensure we're on a stable, compliant EKS version as we transition to VAEC—reducing the chance of disruptions and issues. No challenges or risks at this time. |
| 1/17/25 | In-progress | End of Q1 2025 | Sprint 19: 2 tickets are planned for the current sprint. While the [EVSS PCIU Migration](https://github.com/department-of-veterans-affairs/va.gov-team/issues/73776) and the [EVSS Migration and Deprecation](https://github.com/department-of-veterans-affairs/va.gov-team/issues/100351) have been identified as top priorities, Platform work is blocked on PCIU pending the mobile migration being complete and 1 developer is focusing on the EVSS migration discovery. So work was able to resume on the EKS upgrade. On 1/16 it came to my attention that the EKS Upgrade is a dependency with the VAEC initiative, so the my current challenge is figuring out what those dependencies are. |
| 1/10/25 | In-progress | End of Q1 2025 | Sprint 18: Work on this initiative is currently paused in favor of a higher priority initiative, the [EVSS PCIU Migration](https://github.com/department-of-veterans-affairs/va.gov-team/issues/73776). Also, the [EVSS Migration and Deprecation](https://github.com/department-of-veterans-affairs/va.gov-team/issues/100351) has been identified as a high priority, so additional resources have shifted over to cover the discovery and design documentation. Work is planned to resume in Sprint 19 or 20, depending on how the work can be slotted in taking into account higher priorities. The current challenge is needing to delay work given the urgent timing of the PCIU migration and additional EVSS Migration. |
| 1/3/25 | In-progress | End of Q1 2025 | Sprint 18: Work on this initiative is currently paused in favor of a higher priority initiative, the [EVSS PCIU Migration](https://github.com/department-of-veterans-affairs/va.gov-team/issues/73776). Work is planned to resume in Sprint 19. The current challenge is needing to delay work given the urgent timing of the PCIU migration and additional EVSS dependencies. |
| 12/20/24 | Paused | End of Q1 2025 | Sprint 17: Work on this initiative is currently paused in favor of a higher priority initiative, the [EVSS PCIU Migration](https://github.com/department-of-veterans-affairs/va.gov-team/issues/73776). Work is planned to resume in Sprint 19. The current challenge is needing to delay work given the urgent timing of the PCIU migration and additional EVSS dependencies. |
| 12/13/24 | In-progress | Est. completion early Q1 2025 | Sprint 16: A test plan is finalized, which will verify that all types of login work as expected. Testing will begin next sprint. No challenges or risks identified at this time. |
| 12/6/24 | In-progress | Est. completion early Q1 2025 | Sprint 16: A test plan is finalized, which will verify that all types of login work as expected. Testing will begin next sprint. No challenges or risks identified at this time. |
| 11/22/24 | In-progress | On-track (dependency resolved). Est. completion early Q1 2025  | The dependency has been resolved (2 of 2 dependent issues are complete - [#95636](https://github.com/department-of-veterans-affairs/va.gov-team/issues/95636) and [#95631](https://github.com/department-of-veterans-affairs/va.gov-team/issues/95631)). No follow-up work related to the unauthenticated endpoint testing is needed at this time. This gets the initiative one step closer to EKS cluster testing. Sprint 15: Discovery is in progress to identify what types of test could be added to verify all of the types of login work. |
| 11/8/24 | In-progress | Off-track (dependency). Est. completion early Q1 2025 | Dependent on Reliability team. 1 of 2 dependent issues were completed (#95636 [complete] and #95631 [incomplete]). When the second one is completed, we can continue.  |
| 10/25/24 | In-progress | Off-track (dependency). Est. completion early Q1 2025 | dependent on Reliability team. 2 (#95636 and #95631) new issues created for them to complete before we can continue.  |

### PROBLEM STATEMENT
Infra is working to upgrade EKS (due end of January 2024). Extensive testing is needed for this upgrade, and the Infra team needs help from us as we have extensive knowledge re vets-api on our team.

### ADDITIONAL INFO
[Testing plan ideas](https://vfs.atlassian.net/wiki/spaces/TT1/pages/2837872659/Testing+Plan+for+EKS+Upgrade) (started by Lindsey)
Testing will will occur in the new EKS clusters. 
This team is happening in collaboration with the infrastructure team (Nate Peterson is the PM contact).

### ACTION ITEMS
1. Review [config map](https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests) in each cluster. When a new version is upgraded, some of the config has to be redone, so this will be good to go over together once they are ready.
2. Review [argo template](https://argo-vetstest-next-eks.vfs.va.gov/login?return_url=https%3A%2F%2Fargo-vetstest-next-eks.vfs.va.gov%2Fapplications)
3. Test (per infra team's request) in **new** EKS cluster [this will be send via request in the engineering workstream shared slack channel]. This will need to be done in dev-api.gov... this will be the new cluster
4. Test fwd proxy
5. test actual service
6. Respond to infra team and confirm reciept of communication

*Anytime the infra team requests testing of vets-api, Rachal, Gia, and Ryan will sync to discuss:
- testing request
- information to review prior to testing
- the testing process that will be followed (this will depend on the testing request)

**DevOps Stuff for Platform Team: Check with Kshitiz and Chris first if this is needed**
- IAM changes (changing the repo)
- The remote state needs to be managed as part of managing the resources

## OCTO Objectives:
**Objective:**
02: Our platforms are the best way to deliver products at VA
**OKR**
OKR1: Our platforms hit the "elite" level (as defined by DORA) on Deployment Frequency, Lead Time for Changes, Change Failure Rate, and Time to Restore Service

### DEFINITION OF DONE
We are confident in flipping the switch to a new EKS cluster!

### VALIDATION

```[tasklist]
### Associated Tickets
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/70146
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74825
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74643
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74655
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/75623
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74637
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/82426
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/77350
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/82296
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/87848
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74633
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74550
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/71315
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74644
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74647
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74824
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74648
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74651
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74653
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74654
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74656
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/91587
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/91593
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/91595
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/91591
```

```[tasklist]
### Post MVP Tickets
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/74649
```

**Comment by laineymajor** [comment]
This will continue as we receive more information from the infra team

**Comment by jennb33** [comment]
Jennifer poked Nate Peterson (who was OOO on 1/10/2024) for Infra bandwith re testing on 1/11/2024

**Comment by jennb33** [comment]
@RachalCassity are there individual work tickets that can be associated to this Epic? I removed the 5 story points to the Epic, because Epics aren't usually pointed, the stories within them are. Let me know if I should write a ticket to associate here. Thanks!

**Comment by LindseySaari** [comment]
After our meeting yesterday, @flooose and @Kshitiz-devops worked with the Infrastructure/Lights team to work on a way to port forward to the vets-next application. Next up is to get the Elasticache/Redis connection working and we should be able to continue to move forward with the testing phase.

**Comment by LindseySaari** [comment]
[I sent this doc](https://dsva.slack.com/archives/C057Y8C9F7X/p1709924225501079) over to IST for review

**Comment by AshleyGuerrant** [comment]
Additional testing if necessary will be performed by the Infrastructure Services Team, or updated requirements can be handed off to the SRE Team.

This epic is stale and was written over 1 year ago. The problem statement references an EKS upgrade that was planned for January 2024. Given the EKS upgrade work completed in the [Automate EKS Cluster Deployment and Tear Down epic](https://github.com/department-of-veterans-affairs/va.gov-team/issues/50671), the remaining 5 tickets in this epic are no longer relevant and have been closed.

Closing out this epic, and a new epic can be drafted if necessary.

cc @jwoodman5 @EWashb
